apps/file_publisher/lib/file_publisher/file_publisher.ex

```diff
@@ -41,3 +34,9 @@ defmodule FilePublisher do
     inspect(message)
   end
 end
+
+defimpl Core.PublisherProtocol, for: Atom do
+  def handle_message(FilePublisher, message) do
```

Comment:
Maybe I don't understand it enough, but how does this satisfy `@spec handle_message(atom(), struct())` for dialyzer?

Reply:
`FilePublisher` is atom, message is struct. :smile:

Reply:
In elixir atom is nearly anything. :smile:  Module names, map keys (can be, but don't need to be). They can be in form of `:atom` or in map keys as `key: <value>` where `key` is Atom so `key == :key`. Ofc, `true` and `false` are also atoms. :smile: